tools/util/include/cutlass/util/device_memory.h

```diff
@@ -56,13 +56,14 @@ T* allocate(size_t count = 1) {
   T* ptr = 0;
 
 #if defined(CUTLASS_ENABLE_SYCL)
-  ptr = syclcompat::malloc<T>(count);
-  if (ptr == nullptr) {
-    throw std::runtime_error("Failed to allocate memory");
+  if (count > 0) {
+    ptr = syclcompat::malloc<T>(count);
```

Comment:
I think it's better if we follow CUDA's approach for allocating memory without using the data type here.
```suggestion
    ptr = reinterpret_cast<T>(syclcompat::malloc(bytes));
```